pkg/controllers/resources/persistentvolumes/syncer.go

```diff
@@ -145,8 +145,15 @@ func (s *persistentVolumeSyncer) Sync(ctx *synccontext.SyncContext, pObj client.
 	if err != nil {
 		return ctrl.Result{}, err
 	} else if !sync {
-		ctx.Log.Infof("delete virtual persistent volume %s, because there is no virtual persistent volume claim with that volume", vPersistentVolume.Name)
-		return ctrl.Result{}, ctx.VirtualClient.Delete(ctx.Context, vObj)
+		if vPersistentVolume.Spec.PersistentVolumeReclaimPolicy == corev1.PersistentVolumeReclaimDelete {
+			ctx.Log.Infof("delete virtual persistent volume %s, because there is no virtual persistent volume claim with that volume", vPersistentVolume.Name)
+			return ctrl.Result{}, ctx.VirtualClient.Delete(ctx.Context, vObj)
+		} else {
```

Comment:
If you've put a return statement in the end of if-block, then else statement is not required. You can put "else code" outside the condition block.